File: stack/dailyTemperatures.go
package main

func dailyTemperatures(temperatures []int) []int {
	res := make([]int, len(temperatures))
	stack := make([][]int, 0)
	for index, item := range temperatures {
		for len(stack) != 0 && stack[len(stack)-1][0] < item {
			_, si := stack[len(stack)-1][0], stack[len(stack)-1][1]
			res[si] = index - si
			stack = stack[:len(stack)-1]
		}
		stack = append(stack, []int{item, index})

	}
	return res
}
func main() {
	dailyTemperatures([]int{73, 74, 75, 71, 69, 72, 76, 73})
}

File: stack/dailyTemperatures.py
class Solution(object):
    def dailyTemperatures(self, temperatures):
        """
        :type temperatures: List[int]
        :rtype: List[int]
        """
        res = [0]*len(temperatures)
        stack = []
        for index, temp in  enumerate(temperatures):
            while stack and temp> stack[-1][0]:
                stackT, stackI = stack.pop()
                res[stackI] = index-stackI
            stack.append([temp, index])
            print(stack)
        print(res)



        
temperatures = [73,74,75,71,69,72,76,73]
Solution().dailyTemperatures(temperatures)